pkg/auth/providers/azure/clients/ms_graph_client.go

```diff
@@ -299,9 +248,49 @@ func (c AccessTokenCache) Export(cache cache.Marshaler, key string) {
 	}
 }
 
-func NewMSGraphClientFromAuthConfig(config *v32.AzureADConfig, cache cache.ExportReplace) (*azureMSGraphClient, error) {
+// NewMSGraphClient returns a client of the Microsoft Graph API. It attempts to get an access token to the API.
+// It first tries to fetch the token from the refresh token, if the access token is found in the database.
+// If that fails, it tries to acquire it directly from the auth provider with the credential (application secret in Azure).
+func NewMSGraphClient(config *v32.AzureADConfig, cache cache.ExportReplace) (*azureMSGraphClient, error) {
```

Comment:
Nit:

Since the cache is only used here, it may make sense for you to accept a secret interface and then construct the cache as part of this method call. That being said, this isn't a big deal since you should be using the constructor in the client rather than calling this directly.

Reply:
Resolving, change made.